src/sentry/models/importchunk.py

```diff
@@ -0,0 +1,136 @@
+
+    # A JSON object map from original pks in the source blob to the pks they were "merged" with (ie,
+    # existing database data that was kept in their stead).
+    existing_map = models.JSONField(default=dict)
+
+    # A JSON object map from original pks in the source blob to the pks they "overwrote" (ie, the data from the model was imported into an existing model, and that model's pk was retained).
+    overwrite_map = models.JSONField(default=dict)
+
+    # If the inserted model has a "slug" field, or some other similar globally unique string
+    # identifier, save it in a map from original pks in the source blob to that "slug" value.
+    inserted_identifiers = models.JSONField(default=dict)
+
+    __repr__ = sane_repr("import_uuid", "model", "min_ordinal", "max_ordinal")
+
+    class Meta:
+        abstract = True
+
+
+@region_silo_only_model
+class RegionImportChunk(BaseImportChunk):
+    """
+    Records the pk mapping for the successful import of instances of a model that lives in the
+    region silo.
+    """
+
+    __relocation_scope__ = RelocationScope.Excluded
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_regionimportchunk"
+
+
+@control_silo_only_model
+class ControlImportChunk(BaseImportChunk):
+    """
+    Records the pk mapping for the successful import of instances of a model that lives in the
+    control silo.
+    """
+
+    __relocation_scope__ = RelocationScope.Excluded
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_controlimportchunk"
+
+
+@region_silo_only_model
+class ControlImportChunkReplica(BaseImportChunk):
+    """
+    A duplicate of a `ControlImportChunk` saved on the control silo. Unlike other `*Replica` models,
+    this replica is NOT generated via outbox synchronization; rather, it is the responsibility of
+    the import algorithm to create this duplicate as soon as the RPC call to the control silo
+    returns.
+
+    The purpose of this model is to ensure that the `sentry.backup._import()` method, which always
+    executes from the region silo and is responsible for orchestrating the entire import, has an
+    accurate view of the current import state. For models housed in the region silo, this is simple:
+    we just atomically write a `RegionImportChunk` for the models being imported in the local atomic
+    transaction. For models housed in the control silo, things get more complex: we must make an RPC
+    call and, when writing the models atomically in that silo, create a `ControlImportChunk`. But
+    the `ControlImportChunk` only lives in the control silo, creating the possibility of potentially
+    racy retrieval calls if the `_import()` orchestrator tries to access it directly.
+
+    To resolve this, as soon as the RPC call returns, we have the orchestrator make its own,
+    region-silo housed copy of the `ControlImportChunk`, called `ControlImportChunkReplica`. Thus,
+    when importing a control model, we have 3 scenarios:
```

Comment:
How will timeouts work in this flow? RPC calls can timeout during the response generation, and the client will not be able to see whether or not the operation succeeded. Will you be able to retry the chunk creation/replication again with idempotency?

Reply:
Yep, you will. The follow on PR that implements the features enabled by this model have some tests to ensure that this is the case: https://github.com/getsentry/sentry/pull/59879/files#diff-63a574769742ecc2d07aaaa9c62b96220cfd1055537e72db3a450808cd7a24bdR36